Sources/NIOCore/AsyncSequences/NIOAsyncWriter.swift

```diff
@@ -692,15 +679,18 @@ extension NIOAsyncWriter {
                 inDelegateOutcall: Bool,
                 cancelledYields: [YieldID],
                 suspendedYields: _TinyArray<SuspendedYield>,
-                elements: Deque<Element>,
                 delegate: Delegate
             )
 
-            /// The state once the writer finished and there are still elements that need to be delivered. This can happen if:
+            /// The state once the writer finished and there are still task that need to write. This can happen if:
             /// 1. The ``NIOAsyncWriter`` was deinited
             /// 2. ``NIOAsyncWriter/finish(completion:)`` was called.
             case writerFinished(
-                elements: Deque<Element>,
+                isWritable: Bool,
+                inDelegateOutcall: Bool,
+                suspendedYields: _TinyArray<SuspendedYield>,
+                cancelledYields: [YieldID],
+                knownYieldIDs: _TinyArray<YieldID>,
```

Comment:
What are the known yield IDs?

Reply:
I added a comment to clarify this but they are the buffered yields before we got the finish.